src/printer_yin.c

```diff
@@ -0,0 +1,2439 @@
+
+    /* revision-stmts */
+    if (module->revision) {
+        ly_print(ctx->out, "\n%*srevision %s;\n", INDENT, module->revision);
+    }
+
+    LEVEL--;
+    ly_print(ctx->out, "%*s}\n", INDENT);
+    ly_print_flush(ctx->out);
+
+    return LY_SUCCESS;
+}
+
+static void
+ypr_xmlns(struct ypr_ctx *ctx, const struct lys_module *module)
+{
+    unsigned int u;
+    char *space = "        ";
+
+    ly_print(ctx->out, "%s%*sxmlns=\"%s\"", space, INDENT, YIN_NS_URI);
+    ly_print(ctx->out, "\n%s%*sxmlns:%s=\"%s\"", space, INDENT, module->prefix, module->ns);
+
+    struct lysp_module *modp = module->parsed;
+
+    LY_ARRAY_FOR(modp->imports, u){
+        ly_print(ctx->out, "\n%s%*sxmlns:%s=\"%s\"", space, INDENT, modp->imports[u].prefix, modp->imports[u].module->ns);
+    }
+}
+
+
+struct ext_substmt_info_s stmt_attr_info[] = {
+    {NULL,               NULL,          0},              /**< LY_STMT_NONE*/
+    {"status",           "value",       SUBST_FLAG_ID}, /**< LY_STMT_STATUS */
+    {"config",           "value",       SUBST_FLAG_ID}, /**< LY_STMT_CONFIG */
+    {"mandatory",        "value",       SUBST_FLAG_ID}, /**< LY_STMT_MANDATORY */
+    {"units",            "name",        SUBST_FLAG_ID}, /**< LY_STMT_UNITS */
+    {"default",          "value",       SUBST_FLAG_ID}, /**< LY_STMT_DEFAULT */
+    {"type",             "name",        SUBST_FLAG_ID}, /**< LY_STMT_TYPE */
+    {"action",           "name",        SUBST_FLAG_ID}, /**< LY_STMT_ACTION */
+    {"anydata",          "name",        SUBST_FLAG_ID}, /**< LY_STMT_ANYDATA */
+    {"anyxml",           "name",        SUBST_FLAG_ID}, /**< LY_STMT_ANYXML */
+    {"argument",         "name",        SUBST_FLAG_ID}, /**< LY_STMT_ARGUMENT */
+    {"augment",          "target-node", SUBST_FLAG_ID}, /**< LY_STMT_AUGMENT */
+    {"base",             "name",        SUBST_FLAG_ID}, /**< LY_STMT_BASE */
+    {"belongs-to",       "module",      SUBST_FLAG_ID}, /**< LY_STMT_BELONGS_TO */
+    {"bit",              "name",        SUBST_FLAG_ID}, /**< LY_STMT_BIT */
+    {"case",             "name",        SUBST_FLAG_ID}, /**< LY_STMT_CASE */
+    {"choice",           "name",        SUBST_FLAG_ID}, /**< LY_STMT_CHOICE */
+    {"contact",          "text",        SUBST_FLAG_YIN},/**< LY_STMT_CONTACT */
+    {"container",        "name",        SUBST_FLAG_ID}, /**< LY_STMT_CONTAINER */
+    {"description",      "text",        SUBST_FLAG_YIN},/**< LY_STMT_DESCRIPTION */
+    {"deviate",          "value",       SUBST_FLAG_ID}, /**< LY_STMT_DEVIATE */
+    {"deviation",        "target-node", SUBST_FLAG_ID}, /**< LY_STMT_DEVIATION */
+    {"enum",             "name",        SUBST_FLAG_ID}, /**< LY_STMT_ENUM */
+    {"error-app-tag",    "value",       SUBST_FLAG_ID}, /**< LY_STMT_ERROR_APP_TAG */
+    {"error-message",    "value",       SUBST_FLAG_YIN},/**< LY_STMT_ERROR_MESSAGE */
+    {"extension",        "name",        SUBST_FLAG_ID}, /**< LY_STMT_EXTENSION */
+    {"feature",          
```

Comment:
This is supposed to be XML element and XML comment.

Reply:
Also, I don't understand this part.

Reply:
You print "extension_name; // Model comes from different input..." which is not XML. The `extension_name` is supposed to be XML element and that comment is supposed to be XML comment (`<!-- ... -->`)

Reply:
Note that the code has changed, so github does not show the code snippet correctly in Conversation tab, open  the `Files changed" tab.

Reply:
Oh...I just read in the conversation tab so I don't understand! Now I get it. Thanks